prometheus/src/main/scala/filodb/prometheus/ast/Expressions.scala

```diff
@@ -41,10 +43,22 @@ trait Expressions extends Aggregates with Functions {
     // scalastyle:off method.length
     // scalastyle:off cyclomatic.complexity
     override def toSeriesPlan(timeParams: TimeRangeParams): PeriodicSeriesPlan = {
-      if (hasScalarResult(lhs) && hasScalarResult(rhs)) {
+      val lhsWithPrecedence = lhs match {
+       case p: PrecedenceExpression  => p.expression
+       case _                        => lhs
+
+     }
+
+      val rhsWithPrecedence = rhs match {
+        case p: PrecedenceExpression  => p.expression
+        case _                        => rhs
+
+      }
```

Comment:
Why don't you fold this logic into hasScalarResult ?

Reply:
As discussed we need it for subsequent case statements